src/utils/jupiter-fs.ts

```diff
@@ -91,47 +138,102 @@ export default function JupiterFs({
           encryptSecret,
           feeNQT,
         }
+
+
+        console.log(`sending a record to `, this.client.address);
         await this.client.storeRecord(newAddyInfo, SUBTYPE_MESSAGING_METIS_METADATA)
-        addy = newAddyInfo
+        console.log(`record sent `);
+        binaryAccountInfo = newAddyInfo
       }
-      await this.checkAndFundAccount(addy.address, minimumFndrAccountBalance)
-      this.binaryClient = JupiterClient({ ...addy,
+
+      // console.log(`funding the account `, binaryAccountInfo.address , minimumFndrAccountBalance)
+      // console.log('funder=',  this.jupiterClientOptions.address   )
+      console.log(`jupiterfs().getOrCreateBinaryAddress().checkAndFundAccount(binaryAddress=${binaryAccountInfo.address}, minBalance=${minimumFndrAccountBalance},fundingAmount=${fundingAmount})`)
+      await this.checkAndFundAccount(binaryAccountInfo.address, minimumFndrAccountBalance, fundingAmount)
+      // console.log(`checkAndFundAccount DONE`);
+
+
+      this.binaryClient = JupiterClient({ ...binaryAccountInfo,
         server: jupServer,
         feeNQT,
         minimumFndrAccountBalance,
         minimumUserAccountBalance })
-      return addy
+      return binaryAccountInfo
     },
 
-    async checkAndFundAccount(targetAddress: string, minBalance: number) {
-      const minBalanceBI = new BigNumber(minBalance)
+    /**
+     *
+     * @param targetAddress  ie Binary Account
+     * @param minimumTargetBalance
+     */
+    async checkAndFundAccount(targetAddress: string, minimumTargetBalance: number, fundingAmount: number) {
+      console.log('######################################')
```

Comment:
we should remove console.log

Reply:
@joramirezStackit i think just ignore all of jupiter-fs. this code will be deleted and replaced by the library.